components/studio/bin/hab-studio.sh

```diff
@@ -1076,6 +1076,35 @@ chown_artifacts() {
   fi
 }
 
+# **Internal**  Run a command which may fail without aborting whole script
+try() {
+  local status
+  if [[ $- == *e* ]]; then
+    # Disable `set -e` as briefly as possible to avoid masking other errors
+    set +e
+    $*
+    status=$?
+    set -e
+  else
+    $*
+    status=$?
+  fi
```

Comment:
Would something like this work to avoid the `set -e` check?

```sh
if $*; then
  status=$?
else
  status=$?
fi
```

Reply:
It's strange; I think this was my first approach, but I couldn't get it working right. I tried again and it seems correct. I changed `$*` to `"$@"` to make `shellcheck` happy, but it seems to work correctly in `sh`:
```sh
➤ sh -l
$ good() { return 0; }
$ bad()  { return 7; }
$ try() {
>     if "$@"; then
>         status=$?
>     else
>         status=$?
>         echo "Warning: '$*' failed with status $status"
>     fi
> }
$ try true
$ try false
Warning: 'false' failed with status 1
$ try touch foo
$ try touch /nonexist/bar
touch: cannot touch '/nonexist/bar': No such file or directory
Warning: 'touch /nonexist/bar' failed with status 1
```

Thanks for pointing this out; it's much nicer!

Reply:
I originally was trying a variant of `if ! $*` but I think that flipped the exit code for the body of the conditional. Interesting stuff